Issue 86 - Remover GitHub token quando desejado

cantuariavc opened the pull request:
Signed-off-by: Vinícius Cantuária <cantuariavc@gmail.com>

## Mudanças Propostas

Remove do banco de dados o token do GitHub quando desejado.

## Tipo de Mudança

Que tipo de mudanças este Pull Request introduz ao Falko?
_Marque um `x` ao que se aplicar_

- [x] Conserto de Bug
- [ ] Nova Feature

## Checklist

_Marque um `x` ao que se aplicar. Se você não tiver certeza sobre algum dos tópicos, não hesite em perguntar. Estamos aqui para ajudar!_

- [x] O nome do pull request deve ser auto-explicativo e deve estar em português.
- [x] Os commits seguem a [política de commits do repositório](https://github.com/fga-gpp-mds/Falko-2017.2-BackEnd/wiki/Plano-de-Gerenciamento-de-Configura%C3%A7%C3%A3o-de-Software#41---pol%C3%ADtica-de-commits).
- [x] Não há erros de linter.
- [x] A build está passando (testes, code climate e codacy).
- [x] O pull request está linkado à uma issue existente.

## Screenshots
![captura de tela 2017-11-14 as 02 51 56](https://user-images.githubusercontent.com/21143590/32763441-d15c5d68-c8e6-11e7-8f3e-0d9c7fdaba5f.png)
![captura de tela 2017-11-14 as 02 52 05](https://user-images.githubusercontent.com/21143590/32763440-d136fb18-c8e6-11e7-84e1-71e1a6d8e4f0.png)

## Outros Comentários
Se for uma mudança relativamente grande/complexa e você tiver algum comentário/pergunta a fazer, explicação de sua escolha da sua solução ou qualquer coisa que ache relevante, não hesite em compartilhar.

drianne committed:
Removing GitHub token

Signed-off-by: Vinícius Cantuária <cantuariavc@gmail.com>

cantuariavc commented:
Ainda tem um bug. Tem que sair do componente User para atualizar.

MatheusRich commented:
@cantuariavc ao invés de ter dois botões aparecendo, deveria haver um v-if / v-else para aparecer renderizar apenas um dos botões.

alaxalves commented:
@cantuariavc faz isso que o @MatheusRich falou e desaparece com esse booleano.
![image](https://user-images.githubusercontent.com/19597045/32781433-3138ffe4-c92c-11e7-94bf-b40001c3f62f.png)

drianne committed:
Removing the double rendering of the link button

Signed-off-by: Vinícius Cantuária <cantuariavc@gmail.com>

cantuariavc commented:
Pronto.

alaxalves commented:
@cantuariavc Manda os prints de tela com as atualizações que vc fez!

cantuariavc commented:
![captura de tela 2017-11-15 as 15 35 56](https://user-images.githubusercontent.com/21143590/32850976-dca1caf2-ca1a-11e7-96c0-003da3805739.png)
![captura de tela 2017-11-15 as 15 36 02](https://user-images.githubusercontent.com/21143590/32850977-dcc7cacc-ca1a-11e7-84c4-52a7b0322f48.png)
![captura de tela 2017-11-15 as 15 36 06](https://user-images.githubusercontent.com/21143590/32850978-dcf1b7a6-ca1a-11e7-87cb-1c5710ffd7ac.png)
![captura de tela 2017-11-15 as 15 36 11](https://user-images.githubusercontent.com/21143590/32850979-dd144f3c-ca1a-11e7-942e-73cfb4010bbb.png)

drianne committed:
Fixing releoad bug

Signed-off-by: Vinícius Cantuária <cantuariavc@gmail.com>

cantuariavc committed:
Replacing setting local storage of token

cantuariavc commented:
![captura de tela 2017-11-16 as 14 33 57](https://user-images.githubusercontent.com/21143590/32903093-4461c212-cadb-11e7-8353-19fce913da83.png)
![captura de tela 2017-11-16 as 14 29 51](https://user-images.githubusercontent.com/21143590/32903096-44aec45e-cadb-11e7-99d3-9fa2f94bd8f3.png)

cantuariavc commented:
@alaxalves, @MatheusRich e @matheussbernardo, no meu pc os testes passam.

cantuariavc committed:
Decreasing code complexity and fixing register

cantuariavc committed:
Merging branch 'devel' into issue_86_tirar_github_link

cantuariavc committed:
Fixing tests

ThalissonMelo commented:
Ficou muito bom!
só vou fazer uma pergunta, é necessário deixar o botão de importar repositório mesmo quando o usuário não está linkado? sei q ele fica cinza e apagado mas nessa tela só tem dois botões... talvez seja desnecessário.
A solução do booleano no local storage ficou elegante, mas vamos parar por aí pra não poluir de mais esse espaço kkkk

MatheusRich commented:
@cantuariavc você deixou ambos os botões na tela de Perfil?

cantuariavc commented:
Acho melhor deixar, @ThalissonMelo.

cantuariavc commented:
Não, @MatheusRich.

cantuariavc committed:
Merging branch 'devel'

Conflicts:
	src/components/Authentication/Login.vue
	src/components/Authentication/Register.vue
	src/components/GitHub/GitHubCallBack.vue
	src/components/Projects/Projects.vue
	src/components/Users/UserProfile.vue
	test/unit/specs/GitHubCallBack.spec.js
	test/unit/specs/GitHubProjects.spec.js

alaxalves commented:
Vou aceitar @cantuariavc mas lembre-se de refatorar os tratamentos de estados posteriormente usando o VueX

cantuariavc commented:
Resolve a issue #86
